File: frontend/multilang_examples/hello.scala
object HelloWorld {
  def greet(name: String): String = s"Hello, $name!"
  def add(a: Int, b: Int): Int = a + b
  def main(args: Array[String]): Unit = {
    println(greet("World"))
    println(s"Sum: ${add(3, 5)}")
  }
}

File: frontend/multilang_examples/hello.go
package main
import "fmt"
func greet(name string) string {
    return fmt.Sprintf("Hello, %s!", name)
}
func add(a, b int) int {
    return a + b
}
func main() {
    fmt.Println(greet("World"))
    fmt.Println("Sum:", add(3, 5))
}
